### PR TITLE
Bugfix etb#178 fixing async rx

### PR DESCRIPTION
**Pull request by jrachucki-n7s**

### CODE REVIEW

File: src/samv71_serial_ccsds/samv71_serial_ccsds.c [hunk 1]
```diff
@@ -277,17 +320,32 @@ void SamV71SerialCcsdsPoll(void *private_data) {
...
   while (true) {
-    Uart_read(&self->m_hal_uart.uart, self->m_recv_buffer, UINT32_MAX,
-              &errorCode);
-    if (errorCode == Uart_ErrorCodes_Timeout) {
+    /// Wait for data to arrive. Semaphore will be given
+    xSemaphoreTake(self->m_rx_semaphore, portMAX_DELAY);
+
+    size_t recvBytesCount = 0;
+    for (recvBytesCount = 0;
+         (recvBytesCount < Serial_CCSDS_SAMV71_RECV_BUFFER_SIZE) |
```
Comment (rphh) [inline src/samv71_serial_ccsds/samv71_serial_ccsds.c]: bit or used instead of logical or?

File: src/samv71_serial_ccsds/samv71_serial_ccsds.c [hunk 2]
```diff
@@ -277,17 +320,32 @@ void SamV71SerialCcsdsPoll(void *private_data) {
...
+         (recvBytesCount < Serial_CCSDS_SAMV71_FIFO_BUFFER_SIZE);
```
Comment (rphh) [inline src/samv71_serial_ccsds/samv71_serial_ccsds.c]: Why second comparison is necessary?

**Review by rphh — CHANGES_REQUESTED** [review]

File: src/samv71_serial_ccsds/samv71_serial_ccsds.c (hunk 1 above)
Comment (jrachucki-n7s) [reply src/samv71_serial_ccsds/samv71_serial_ccsds.c]: This was a typo. It is fixed in new commit

File: src/samv71_serial_ccsds/samv71_serial_ccsds.c (hunk 2 above)
Comment (jrachucki-n7s) [reply src/samv71_serial_ccsds/samv71_serial_ccsds.c]: I realize it is not neccessary. It was deleted in new commit.

**Review by rphh — APPROVED** [review]

**Pull request merged**